In [2]:
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.datasets import load_iris

iris = load_iris()

In [4]:
dir(iris)

['DESCR',
 'data',
 'data_module',
 'feature_names',
 'filename',
 'frame',
 'target',
 'target_names']

In [6]:
from sklearn.model_selection import cross_val_score

In [18]:
a = cross_val_score(LogisticRegression(),iris.data, iris.target)
div = len(a)
res = 0
for i in a:
    res+=i

avg = res/div
print(avg)

0.9733333333333334


C:\Users\yoges\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:469: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [20]:
a = cross_val_score(SVC(),iris.data, iris.target)
div = len(a)
res = 0
for i in a:
    res+=i

avg = res/div
print(avg)

0.9666666666666666


In [22]:
a = cross_val_score(RandomForestClassifier(n_estimators=10),iris.data, iris.target)
div = len(a)
res = 0
for i in a:
    res+=i

avg = res/div
print(avg)

0.9466666666666667


#### SVM is the best performing model here for the Iris dataset on evaluating with the K-Fold Cross Validation with an accuracy of 96.7%

In [27]:
import numpy as np
a = cross_val_score(SVC(),iris.data, iris.target)
print(np.average(a))

0.9666666666666666


In [31]:
b = cross_val_score(LogisticRegression(),iris.data, iris.target)
print(np.average(b))

0.9733333333333334


C:\Users\yoges\anaconda3\Lib\site-packages\sklearn\linear_model\_logistic.py:469: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [33]:
import numpy as np
c = cross_val_score(RandomForestClassifier(n_estimators=10),iris.data, iris.target)
print(np.average(c))

0.9533333333333334


#### The best model here when we use the 'average' function of Numpy is I am getting it as 'Logistic Regression'